lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -20,15 +20,187 @@ const getMethodAuthorization = require('./lib/method/authorization');
+}
+
+const contentHandlingSchema = { enum: ['CONVERT_TO_BINARY', 'CONVERT_TO_TEXT'] };
+
+const allowedMethods = ['GET', 'POST', 'PUT', 'PATCH', 'OPTIONS', 'HEAD', 'DELETE', 'ANY'];
+const methodPattern = new RegExp(`^(?:\\*|${allowedMethods.join('|')})$`, 'i');
+const methodPathPattern = new RegExp(`^(?:\\*|(${allowedMethods.join('|')}) (\\/\\S*))$`, 'i');
+
+const requestParametersSchema = {
+  type: 'object',
+  additionalProperties: {
+    oneOf: [
+      { type: 'boolean' },
+      {
+        type: 'object',
+        properties: {
+          required: { type: 'boolean' },
+          mappedValue: { type: 'string' },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+};
+
+const authorizerSchema = {
+  oneOf: [
+    { type: 'string' },
+    caseInsensitive('aws_iam'),
+    { $ref: '#/definitions/awsArnString' },
+    {
+      type: 'object',
+      properties: {
+        arn: { $ref: '#/definitions/awsArn' },
+        authorizerId: { $ref: '#/definitions/awsCfInstruction' },
+        claims: { type: 'array', items: { type: 'string' } },
+        identitySource: { type: 'string' },
+        identityValidationExpression: { type: 'string' },
+        managedExternally: { type: 'boolean' },
+        name: { type: 'string' },
+        resultTtlInSeconds: { type: 'integer', minimum: 0, maximum: 3600 },
+        scopes: { type: 'array', items: { type: 'string' } },
+        type: {
+          oneOf: ['token', 'cognito_user_pools', 'request', 'aws_iam'].map(caseInsensitive),
+        },
+      },
+      required: [],
+      additionalProperties: false,
+    },
+  ],
+};
+
+const corsSchema = {
+  oneOf: [
+    { type: 'boolean' },
+    {
+      type: 'object',
+      properties: {
+        allowCredentials: { type: 'boolean' },
+        cacheControl: { type: 'string' },
+        headers: { type: 'array', items: { type: 'string' } },
+        maxAge: { type: 'integer', minimum: 1 },
+        methods: { type: 'array', items: { enum: allowedMethods } },
+      },
+      oneOf: [
+        { properties: { origin: { type: 'string' } } },
+        {
+          properties: {
+            origins: {
+              type: 'array',
+              items: { type: 'string' },
+            },
+          },
+        },
+      ],
+      additionalProperties: false,
+    },
+  ],
+};
+
+const requestSchema = {
+  type: 'object',
+  properties: {
+    contentHandling: contentHandlingSchema,
+    method: { type: 'string', regexp: methodPattern.toString() },
+    parameters: {
+      type: 'object',
+      properties: {
+        querystrings: requestParametersSchema,
+        headers: requestParametersSchema,
+        paths: requestParametersSchema,
+      },
+      additionalProperties: false,
+    },
+    passThrough: { enum: ['NEVER', 'WHEN_NO_MATCH', 'WHEN_NO_TEMPLATES'] },
+    schema: {
+      type: 'object',
+      additionalProperties: { type: ['object', 'string'] },
```

Comment:
On AWS side it's type _Json_, does it mean that both forms are accepted, _object_ and _string_ ?

Reply:
I got confused by the example:
```yaml
functions:
  create:
    handler: posts.create
    events:
      - http:
          path: posts/create
          method: post
          request:
            schema:
              application/json: ${file(create_request.json)}
```
I understand now that `${file(create_request.json)}` resolves to json (as object) so the `string` option is not relevant